[2017-11-21 01:41:24,596] olympus.Olympus<Olympus:8f002>:OUTPUT: Parsed config: {'clients': {}, 'replicas': {}, 'failures': {0: {'r0': {'client_request': [{'failure': ('invalid_order_sig', ''), 'trigger_point': {'event_num': 2, 'application': 'post', 'client_id': 'c0'}}]}}}, 'test_case_name': 'test_invalid_order_sig_t2_multi_client', 't': 2, 'num_client': 1, 'client_timeout': 3000, 'head_timeout': 3000, 'nonhead_timeout': 3000, 'checkpt_interval': 5, 'hosts': {0: 'localhost;'}, 'client_hosts': {'c0': 'localhost;', 'c1': 'localhost;'}, 'replica_hosts': {'r0': '0', 'r1': '0', 'r2': '0', 'r3': '0', 'r4': '0;'}, 'workloads': {'c0': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"]}}
[2017-11-21 01:41:24,603] olympus.Olympus<Olympus:8f002>:OUTPUT: Generated keys: {'private_keys': {'o1': b'ad316be4f775598edfde61feaf78169f1e5277c9cc55d546d1c9c3c251f09526', 'c0': b'efb33292a38912bb3133d8854b24c9981758aa1b01c51d73ff8d324efc5442fa', 'c1': b'6923adbb65502aaa0dfbef4b68a613e7f5f1756e5855f0551e60fa4eb0670217', 'r0': b'f9b28ee751b2d3c0ede3851c2bfb1883910fad84a338e2a24be89b115afe837b', 'r1': b'2a94c7b775f1af639abf47ecac40693af2d77f975cfface1e2cef0d778f1a79d', 'r2': b'9bb9b314b1234218430f18ec20d73d795b0569f1e5099b292ae185704229e438', 'r3': b'299296a73af98632af16702638c5b9553402504c5c640b1e2c42ba6135a9e663', 'r4': b'a4c9ff4ad8426baf76f1fcf179f235d90addf4b0fc2a70db0196a616bd0fda89'}, 'public_keys': {'o1': b'3a86cbbd8d7ff05c991522c1911c5443845c69e78738faec469b778298eefaae', 'c0': b'1384e1ecff6efd11bd538890f04775a720865b9542b5815bfb0fe4d503d15e32', 'c1': b'198fecb429e9ad41d927090abeb2abb6c9522d4880505b52a1cfaea09df0d065', 'r0': b'e9e1f8ae30d6f310839e79a2772961e2a588443734a78fe22251539a9bad952f', 'r1': b'8828ae116d6005b1af2c34dccd342242722770dcd649cfbef81f66879782c8d0', 'r2': b'ed3fb7ee7f26a2babe2f08efe283616b580419ac27abff33e1eb1ab9624f7ad3', 'r3': b'f4d1523a18e9aab0d190650d574e0d5910da92925df0f765205d3787e25c3dbb', 'r4': b'ee425524526100bd9e8a77f8d021127976350e1621e6fd738b58d9450fbb5243'}}
[2017-11-21 01:41:24,834] olympus.Olympus<Olympus:8f002>:OUTPUT: Initializing cluster with configuration version:  0
[2017-11-21 01:41:26,047] replica.Replica<Replica:a0002>:OUTPUT: Spawned and initialized replica:  r0
[2017-11-21 01:41:26,053] replica.Replica<Replica:a0003>:OUTPUT: Spawned and initialized replica:  r1
[2017-11-21 01:41:26,058] replica.Replica<Replica:a0004>:OUTPUT: Spawned and initialized replica:  r2
[2017-11-21 01:41:26,063] replica.Replica<Replica:a0005>:OUTPUT: Spawned and initialized replica:  r3
[2017-11-21 01:41:26,069] replica.Replica<Replica:a0006>:OUTPUT: Spawned and initialized replica:  r4
[2017-11-21 01:41:26,075] client.Client<Client:a0001>:OUTPUT: c0  initialized with config= {'id': 'c0', 'workload': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"], 'private_key': b'efb33292a38912bb3133d8854b24c9981758aa1b01c51d73ff8d324efc5442fa', 'public_keys': {'o1': b'3a86cbbd8d7ff05c991522c1911c5443845c69e78738faec469b778298eefaae', 'c0': b'1384e1ecff6efd11bd538890f04775a720865b9542b5815bfb0fe4d503d15e32', 'c1': b'198fecb429e9ad41d927090abeb2abb6c9522d4880505b52a1cfaea09df0d065', 'r0': b'e9e1f8ae30d6f310839e79a2772961e2a588443734a78fe22251539a9bad952f', 'r1': b'8828ae116d6005b1af2c34dccd342242722770dcd649cfbef81f66879782c8d0', 'r2': b'ed3fb7ee7f26a2babe2f08efe283616b580419ac27abff33e1eb1ab9624f7ad3', 'r3': b'f4d1523a18e9aab0d190650d574e0d5910da92925df0f765205d3787e25c3dbb', 'r4': b'ee425524526100bd9e8a77f8d021127976350e1621e6fd738b58d9450fbb5243'}, 'client_timeout': 3.0, 'olympus': {'id': 'o1', 'process_id': <Olympus:8f002>}, 'replicas': {'r0': <Replica:a0002>, 'r1': <Replica:a0003>, 'r2': <Replica:a0004>, 'r3': <Replica:a0005>, 'r4': <Replica:a0006>}, 'head_id': 'r0', 't': 2}
[2017-11-21 01:41:26,078] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}
[2017-11-21 01:41:26,079] client.Client<Client:a0001>:OUTPUT: return_value= OK
[2017-11-21 01:41:26,083] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}}, 'signed_content': b"l (Y\xf8\xf5\x0c\xaak~S\xfa\xfa\xb1\x97gJ%\xf2\xd1\xa3\x97t\x8c\xf4<\xd9E\x10e%\xec\xbc6\x02q\xdceN_m\x1d!\x8f\xaf/[\n! \xc8\xff\r_0\xd8\xba\x8fq\r`\xca;\r\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x00X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x13\x00\x00\x00put('movie','star')q\x08uu."}
[2017-11-21 01:41:26,083] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 0 r0
[2017-11-21 01:41:26,084] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:26,085] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:26,088] replica.Replica<Replica:a0002>:OUTPUT: Head is processing request (operation_id, client_id) (0, 'c0')
[2017-11-21 01:41:26,094] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:26,094] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:26,098] replica.Replica<Replica:a0003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 0 r1
[2017-11-21 01:41:26,099] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:26,099] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:26,115] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:26,115] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:26,119] replica.Replica<Replica:a0004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 0 r2
[2017-11-21 01:41:26,120] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:26,120] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:26,135] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:26,135] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:26,140] replica.Replica<Replica:a0005>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 0 r3
[2017-11-21 01:41:26,140] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:26,141] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:26,157] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:26,158] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:26,163] replica.Replica<Replica:a0006>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 0 r4
[2017-11-21 01:41:26,164] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:26,164] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,197] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:26,197] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,203] replica.Replica<Replica:a0005>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 0 r3
[2017-11-21 01:41:26,203] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:26,203] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:26,203] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,204] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:26,208] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:26,226] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:26,226] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:26,227] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"} OK
[2017-11-21 01:41:26,228] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= OK {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}
[2017-11-21 01:41:26,228] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:26,229] client.Client<Client:a0001>:OUTPUT: return_value= star
[2017-11-21 01:41:26,231] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"|>\xd0\x05 d!\n$3\xcb8\xea\xc5\\\x99\xa0\xc9e\xdb=\x01\x10\xb5|\x1f[\x81\x03\x9a\x8e \xb6\xb4\xb2\x82A\x85{Fj\x19]C@\xfaA{`\xbd\xcf\x0e\xd6\xd6F\xe4\xb7'rn\r\xb5\x1a\n\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x01X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:26,232] replica.Replica<Replica:a0004>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 0 r2
[2017-11-21 01:41:26,232] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 1 r0
[2017-11-21 01:41:26,232] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:26,233] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:26,233] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:26,233] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:26,237] replica.Replica<Replica:a0002>:OUTPUT: Head is processing request (operation_id, client_id) (1, 'c0')
[2017-11-21 01:41:26,244] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:26,245] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:26,248] replica.Replica<Replica:a0003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 1 r1
[2017-11-21 01:41:26,249] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:26,249] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:26,254] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:26,254] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:26,264] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:26,265] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:26,267] replica.Replica<Replica:a0003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 0 r1
[2017-11-21 01:41:26,268] replica.Replica<Replica:a0004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 1 r2
[2017-11-21 01:41:26,268] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:26,268] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:26,268] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:26,269] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:26,288] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:26,288] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:26,290] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:26,292] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:26,296] replica.Replica<Replica:a0002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 0 r0
[2017-11-21 01:41:26,297] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:26,297] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:26,298] replica.Replica<Replica:a0005>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 1 r3
[2017-11-21 01:41:26,299] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:26,299] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:26,324] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:26,324] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:26,328] replica.Replica<Replica:a0006>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 1 r4
[2017-11-21 01:41:26,329] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:26,329] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,364] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:26,364] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,368] replica.Replica<Replica:a0005>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 1 r3
[2017-11-21 01:41:26,368] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:26,369] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:26,369] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:26,369] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:26,372] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:26,393] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:26,393] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:26,395] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"} star
[2017-11-21 01:41:26,396] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:26,396] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:26,397] client.Client<Client:a0001>:OUTPUT: return_value= OK
[2017-11-21 01:41:26,397] replica.Replica<Replica:a0004>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 1 r2
[2017-11-21 01:41:26,398] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:26,398] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:26,400] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_content': b'\x17\xa7\xeb\xb2/\r\xefi,\xb1\xba\xb0\x8a\xa48k\xf1\xdb\xb9\x1f\xcf\x17\xb8{\xf4\xb0\xfdR!\x15<\xb5n%\x92p\xf0T%:F\xcb\xd7\x8e\xdeg\xcf\xd3!\x15{\n\xf4\x00\x12j"\xa6f\xacs\x9bg\x0e\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append(\'movie\', \'123\')q\x08uu.'}
[2017-11-21 01:41:26,401] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r0
[2017-11-21 01:41:26,402] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:26,403] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:26,407] replica.Replica<Replica:a0002>:OUTPUT: Head is processing request (operation_id, client_id) (2, 'c0')
[2017-11-21 01:41:26,414] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:26,417] replica.Replica<Replica:a0002>:OUTPUT: Failure invalid_order_sig applied at replica:  r0
[2017-11-21 01:41:26,420] replica.Replica<Replica:a0003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 2 r1
[2017-11-21 01:41:26,420] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:26,421] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:26,422] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:26,422] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:26,425] replica.Replica<Replica:a0003>:OUTPUT: A signature mismatch occurred from source: r0
[2017-11-21 01:41:26,427] replica.Replica<Replica:a0003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 1 r1
[2017-11-21 01:41:26,428] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:26,428] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:26,446] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:26,447] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:26,451] replica.Replica<Replica:a0002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 1 r0
[2017-11-21 01:41:26,452] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:26,452] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:29,402] client.Client<Client:a0001>:OUTPUT: Wait for result timed out for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:29,402] client.Client<Client:a0001>:OUTPUT: Config requested from Olympus by client= c0
[2017-11-21 01:41:29,404] client.Client<Client:a0001>:OUTPUT: Sending config request from client:  c0
[2017-11-21 01:41:29,405] olympus.Olympus<Olympus:8f002>:OUTPUT: Config request received from client= c0
[2017-11-21 01:41:29,410] olympus.Olympus<Olympus:8f002>:OUTPUT: Config has been sent to client= c0
[2017-11-21 01:41:29,413] client.Client<Client:a0001>:OUTPUT: The config was fetched from Olympus: {'id': 'c0', 'workload': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"], 'private_key': b'efb33292a38912bb3133d8854b24c9981758aa1b01c51d73ff8d324efc5442fa', 'public_keys': {'o1': b'3a86cbbd8d7ff05c991522c1911c5443845c69e78738faec469b778298eefaae', 'c0': b'1384e1ecff6efd11bd538890f04775a720865b9542b5815bfb0fe4d503d15e32', 'c1': b'198fecb429e9ad41d927090abeb2abb6c9522d4880505b52a1cfaea09df0d065', 'r0': b'e9e1f8ae30d6f310839e79a2772961e2a588443734a78fe22251539a9bad952f', 'r1': b'8828ae116d6005b1af2c34dccd342242722770dcd649cfbef81f66879782c8d0', 'r2': b'ed3fb7ee7f26a2babe2f08efe283616b580419ac27abff33e1eb1ab9624f7ad3', 'r3': b'f4d1523a18e9aab0d190650d574e0d5910da92925df0f765205d3787e25c3dbb', 'r4': b'ee425524526100bd9e8a77f8d021127976350e1621e6fd738b58d9450fbb5243'}, 'client_timeout': 3.0, 'olympus': {'id': 'o1', 'process_id': <Olympus:8f002>}, 'replicas': {'r0': <Replica:a0002>, 'r1': <Replica:a0003>, 'r2': <Replica:a0004>, 'r3': <Replica:a0005>, 'r4': <Replica:a0006>}, 'head_id': 'r0', 't': 2}
[2017-11-21 01:41:29,417] client.Client<Client:a0001>:OUTPUT: Retransmitting request to all replicas for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:29,420] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r0
[2017-11-21 01:41:29,422] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:29,423] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:29,423] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r1
[2017-11-21 01:41:29,425] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r2
[2017-11-21 01:41:29,426] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r1
[2017-11-21 01:41:29,427] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r2
[2017-11-21 01:41:29,428] replica.Replica<Replica:a0002>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:29,428] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r1
[2017-11-21 01:41:29,427] replica.Replica<Replica:a0005>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r3
[2017-11-21 01:41:29,428] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r2
[2017-11-21 01:41:29,429] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r3
[2017-11-21 01:41:29,430] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r3
[2017-11-21 01:41:29,429] replica.Replica<Replica:a0006>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r4
[2017-11-21 01:41:29,431] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r4
[2017-11-21 01:41:29,432] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r4
[2017-11-21 01:41:29,437] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:29,437] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:29,438] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:29,438] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:29,438] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:29,438] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:29,439] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:29,440] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r2
[2017-11-21 01:41:29,441] replica.Replica<Replica:a0005>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r4 c0 2 r3
[2017-11-21 01:41:29,441] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:29,442] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r3
[2017-11-21 01:41:29,444] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r3
[2017-11-21 01:41:29,444] replica.Replica<Replica:a0002>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:29,447] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:29,448] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:29,449] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:29,451] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:29,451] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:29,452] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:29,452] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:29,452] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r2
[2017-11-21 01:41:29,453] replica.Replica<Replica:a0002>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:29,453] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:29,460] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:29,461] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:29,461] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:29,461] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:29,461] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:29,462] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:29,464] replica.Replica<Replica:a0002>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:29,468] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:29,469] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:29,469] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:29,473] replica.Replica<Replica:a0002>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:32,429] client.Client<Client:a0001>:OUTPUT: Wait for result timed out for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:32,429] client.Client<Client:a0001>:OUTPUT: Config requested from Olympus by client= c0
[2017-11-21 01:41:32,431] client.Client<Client:a0001>:OUTPUT: Sending config request from client:  c0
[2017-11-21 01:41:32,433] olympus.Olympus<Olympus:8f002>:OUTPUT: Config request received from client= c0
[2017-11-21 01:41:32,438] olympus.Olympus<Olympus:8f002>:OUTPUT: Config has been sent to client= c0
[2017-11-21 01:41:32,440] replica.Replica<Replica:a0006>:OUTPUT: Result Shuttle Timeout in Replica:  r4
[2017-11-21 01:41:32,441] client.Client<Client:a0001>:OUTPUT: The config was fetched from Olympus: {'id': 'c0', 'workload': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"], 'private_key': b'efb33292a38912bb3133d8854b24c9981758aa1b01c51d73ff8d324efc5442fa', 'public_keys': {'o1': b'3a86cbbd8d7ff05c991522c1911c5443845c69e78738faec469b778298eefaae', 'c0': b'1384e1ecff6efd11bd538890f04775a720865b9542b5815bfb0fe4d503d15e32', 'c1': b'198fecb429e9ad41d927090abeb2abb6c9522d4880505b52a1cfaea09df0d065', 'r0': b'e9e1f8ae30d6f310839e79a2772961e2a588443734a78fe22251539a9bad952f', 'r1': b'8828ae116d6005b1af2c34dccd342242722770dcd649cfbef81f66879782c8d0', 'r2': b'ed3fb7ee7f26a2babe2f08efe283616b580419ac27abff33e1eb1ab9624f7ad3', 'r3': b'f4d1523a18e9aab0d190650d574e0d5910da92925df0f765205d3787e25c3dbb', 'r4': b'ee425524526100bd9e8a77f8d021127976350e1621e6fd738b58d9450fbb5243'}, 'client_timeout': 3.0, 'olympus': {'id': 'o1', 'process_id': <Olympus:8f002>}, 'replicas': {'r0': <Replica:a0002>, 'r1': <Replica:a0003>, 'r2': <Replica:a0004>, 'r3': <Replica:a0005>, 'r4': <Replica:a0006>}, 'head_id': 'r0', 't': 2}
[2017-11-21 01:41:32,442] replica.Replica<Replica:a0006>:OUTPUT: Sent reconfiguration request to Olympus from : r4
[2017-11-21 01:41:32,443] replica.Replica<Replica:a0002>:OUTPUT: Result Shuttle Timeout in head:  r0
[2017-11-21 01:41:32,445] client.Client<Client:a0001>:OUTPUT: Retransmitting request to all replicas for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:32,447] olympus.Olympus<Olympus:8f002>:OUTPUT: Received reconfig request from:  r4
[2017-11-21 01:41:32,447] olympus.Olympus<Olympus:8f002>:OUTPUT: Received valid reconfiguration request from:  r4
[2017-11-21 01:41:32,450] replica.Replica<Replica:a0006>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r4
[2017-11-21 01:41:32,448] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r1
[2017-11-21 01:41:32,450] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r2
[2017-11-21 01:41:32,450] replica.Replica<Replica:a0005>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r3
[2017-11-21 01:41:32,451] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r1
[2017-11-21 01:41:32,451] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r4
[2017-11-21 01:41:32,452] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r2
[2017-11-21 01:41:32,452] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r3
[2017-11-21 01:41:32,453] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r4
[2017-11-21 01:41:32,452] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r1
[2017-11-21 01:41:32,453] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r2
[2017-11-21 01:41:32,454] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r3
[2017-11-21 01:41:32,454] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent wedge request to:  r0
[2017-11-21 01:41:32,463] replica.Replica<Replica:a0005>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r4 c0 2 r3
[2017-11-21 01:41:32,464] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r3
[2017-11-21 01:41:32,464] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:32,464] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:32,464] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r3
[2017-11-21 01:41:32,465] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r2
[2017-11-21 01:41:32,465] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:32,465] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:32,465] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent wedge request to:  r1
[2017-11-21 01:41:32,466] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:32,469] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent wedge request to:  r2
[2017-11-21 01:41:32,473] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent wedge request to:  r3
[2017-11-21 01:41:32,473] replica.Replica<Replica:a0004>:OUTPUT: Received wedge request from Olympus in Replica:  r2
[2017-11-21 01:41:32,474] replica.Replica<Replica:a0005>:OUTPUT: Received wedge request from Olympus in Replica:  r3
[2017-11-21 01:41:32,474] replica.Replica<Replica:a0003>:OUTPUT: Received wedge request from Olympus in Replica:  r1
[2017-11-21 01:41:32,474] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'pre')  in replica:  r2
[2017-11-21 01:41:32,475] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'pre')  in replica:  r1
[2017-11-21 01:41:32,475] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'pre')  in replica:  r3
[2017-11-21 01:41:32,475] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r2
[2017-11-21 01:41:32,475] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r1
[2017-11-21 01:41:32,476] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r3
[2017-11-21 01:41:32,476] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent wedge request to:  r4
[2017-11-21 01:41:32,477] replica.Replica<Replica:a0006>:OUTPUT: Received wedge request from Olympus in Replica:  r4
[2017-11-21 01:41:32,479] olympus.Olympus<Olympus:8f002>:OUTPUT: Reconfiguration in progress, ignoring reconfig request from: r0
[2017-11-21 01:41:32,479] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'pre')  in replica:  r4
[2017-11-21 01:41:32,480] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r4
[2017-11-21 01:41:32,482] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'post')  in replica:  r2
[2017-11-21 01:41:32,482] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'post')  in replica:  r1
[2017-11-21 01:41:32,483] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'post')  in replica:  r3
[2017-11-21 01:41:32,484] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r2
[2017-11-21 01:41:32,484] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r1
[2017-11-21 01:41:32,484] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r3
[2017-11-21 01:41:32,486] replica.Replica<Replica:a0003>:OUTPUT: Sent Wedged message to Olympus from replica:  r1
[2017-11-21 01:41:32,486] replica.Replica<Replica:a0004>:OUTPUT: Sent Wedged message to Olympus from replica:  r2
[2017-11-21 01:41:32,487] replica.Replica<Replica:a0006>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'post')  in replica:  r4
[2017-11-21 01:41:32,487] replica.Replica<Replica:a0005>:OUTPUT: Sent Wedged message to Olympus from replica:  r3
[2017-11-21 01:41:32,487] replica.Replica<Replica:a0003>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:32,488] replica.Replica<Replica:a0006>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r4
[2017-11-21 01:41:32,488] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:32,488] replica.Replica<Replica:a0004>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:32,489] olympus.Olympus<Olympus:8f002>:OUTPUT: Received wedged message from replica: r2
[2017-11-21 01:41:32,489] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:32,489] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r2
[2017-11-21 01:41:32,490] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:32,491] replica.Replica<Replica:a0006>:OUTPUT: Sent Wedged message to Olympus from replica:  r4
[2017-11-21 01:41:32,483] replica.Replica<Replica:a0002>:OUTPUT: Sent reconfiguration request to Olympus from : r0
[2017-11-21 01:41:32,494] replica.Replica<Replica:a0003>:OUTPUT: Discarded request from  r2  in Replica:  r1  as Replica is in immutable state
[2017-11-21 01:41:32,495] replica.Replica<Replica:a0004>:OUTPUT: Discarded request from  r3  in Replica:  r2  as Replica is in immutable state
[2017-11-21 01:41:32,496] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r0
[2017-11-21 01:41:32,497] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:32,497] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:32,500] replica.Replica<Replica:a0002>:OUTPUT: Head is processing request (operation_id, client_id) (2, 'c0')
[2017-11-21 01:41:32,504] olympus.Olympus<Olympus:8f002>:OUTPUT: Adding the following replica to the quorum: r2
[2017-11-21 01:41:32,505] olympus.Olympus<Olympus:8f002>:OUTPUT: Validated history successfully for wedged message from:  r2
[2017-11-21 01:41:32,505] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:32,506] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:32,507] olympus.Olympus<Olympus:8f002>:OUTPUT: Received wedged message from replica: r1
[2017-11-21 01:41:32,509] replica.Replica<Replica:a0002>:OUTPUT: Received wedge request from Olympus in Replica:  r0
[2017-11-21 01:41:32,510] replica.Replica<Replica:a0003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 2 r1
[2017-11-21 01:41:32,509] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'pre')  in replica:  r0
[2017-11-21 01:41:32,511] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r0
[2017-11-21 01:41:32,516] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('wedge_request', 'post')  in replica:  r0
[2017-11-21 01:41:32,517] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  wedge_request  found in replica:  r0
[2017-11-21 01:41:32,518] olympus.Olympus<Olympus:8f002>:OUTPUT: Adding the following replica to the quorum: r1
[2017-11-21 01:41:32,519] replica.Replica<Replica:a0002>:OUTPUT: Sent Wedged message to Olympus from replica:  r0
[2017-11-21 01:41:32,519] olympus.Olympus<Olympus:8f002>:OUTPUT: Validated history successfully for wedged message from:  r1
[2017-11-21 01:41:32,520] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:32,521] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:32,521] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:32,521] olympus.Olympus<Olympus:8f002>:OUTPUT: Received wedged message from replica: r3
[2017-11-21 01:41:32,525] replica.Replica<Replica:a0002>:OUTPUT: Discarded request from  r1  in Replica:  r0  as Replica is in immutable state
[2017-11-21 01:41:32,526] replica.Replica<Replica:a0002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:32,527] replica.Replica<Replica:a0002>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:32,527] replica.Replica<Replica:a0002>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:32,531] replica.Replica<Replica:a0002>:OUTPUT: Discarded request from  r1  in Replica:  r0  as Replica is in immutable state
[2017-11-21 01:41:32,542] olympus.Olympus<Olympus:8f002>:OUTPUT: Adding the following replica to the quorum: r3
[2017-11-21 01:41:32,543] olympus.Olympus<Olympus:8f002>:OUTPUT: Validated history successfully for wedged message from:  r3
[2017-11-21 01:41:32,543] olympus.Olympus<Olympus:8f002>:OUTPUT: Wedged quorum is satisfied, initiating catch-up process in Olympus.
[2017-11-21 01:41:32,543] olympus.Olympus<Olympus:8f002>:OUTPUT: Finding Longest History from Wedged Messages.
[2017-11-21 01:41:32,544] olympus.Olympus<Olympus:8f002>:OUTPUT: Longest History found in Replica:  r2
[2017-11-21 01:41:32,546] olympus.Olympus<Olympus:8f002>:OUTPUT: Sending catch up message to replica:  r2
[2017-11-21 01:41:32,548] replica.Replica<Replica:a0004>:OUTPUT: Received catch up message from Olympus in = r2
[2017-11-21 01:41:32,549] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'pre')  in replica:  r2
[2017-11-21 01:41:32,549] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r2
[2017-11-21 01:41:32,549] olympus.Olympus<Olympus:8f002>:OUTPUT: Sending catch up message to replica:  r1
[2017-11-21 01:41:32,552] replica.Replica<Replica:a0003>:OUTPUT: Received catch up message from Olympus in = r1
[2017-11-21 01:41:32,552] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'pre')  in replica:  r1
[2017-11-21 01:41:32,553] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r1
[2017-11-21 01:41:32,552] olympus.Olympus<Olympus:8f002>:OUTPUT: Sending catch up message to replica:  r3
[2017-11-21 01:41:32,553] replica.Replica<Replica:a0004>:OUTPUT: Sending caught up message to Olympus from replica:  r2
[2017-11-21 01:41:32,556] replica.Replica<Replica:a0005>:OUTPUT: Received catch up message from Olympus in = r3
[2017-11-21 01:41:32,556] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'post')  in replica:  r2
[2017-11-21 01:41:32,557] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'pre')  in replica:  r3
[2017-11-21 01:41:32,557] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r2
[2017-11-21 01:41:32,557] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r3
[2017-11-21 01:41:32,559] olympus.Olympus<Olympus:8f002>:OUTPUT: Received wedged message from replica: r4
[2017-11-21 01:41:32,561] replica.Replica<Replica:a0003>:OUTPUT: Sending caught up message to Olympus from replica:  r1
[2017-11-21 01:41:32,562] replica.Replica<Replica:a0005>:OUTPUT: Sending caught up message to Olympus from replica:  r3
[2017-11-21 01:41:32,565] replica.Replica<Replica:a0005>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'post')  in replica:  r3
[2017-11-21 01:41:32,565] replica.Replica<Replica:a0003>:OUTPUT: Checking for failure associated with trigger, application:  ('catch_up', 'post')  in replica:  r1
[2017-11-21 01:41:32,565] replica.Replica<Replica:a0005>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r3
[2017-11-21 01:41:32,565] replica.Replica<Replica:a0003>:OUTPUT: No failure associated with trigger:  catch_up  found in replica:  r1
[2017-11-21 01:41:32,585] olympus.Olympus<Olympus:8f002>:OUTPUT: Adding wedged response to wedged_responses: r4
[2017-11-21 01:41:32,586] olympus.Olympus<Olympus:8f002>:OUTPUT: Validated history successfully for wedged message from:  r4
[2017-11-21 01:41:32,588] olympus.Olympus<Olympus:8f002>:OUTPUT: Received wedged message from replica: r0
[2017-11-21 01:41:32,592] olympus.Olympus<Olympus:8f002>:OUTPUT: Received history with holes:  [{'slot': 1, 'operation': {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}, 'order_statements': {'r0': {'order_statement': {'slot': 1, 'operation': {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}}, 'signed_order_statement': b"\x97\n\xd1\xa7%/\xa7a\xc4!]\x17\xc1\r\xc9`\xe1]\xfdN\xd1s\xce\xcaH\xdd=,S\xffQw\x13\x07\xa9\xf1\xc5\xa7\xbf\x81\x11\x98\x06\xa1\x891\x03\x03v\x89'\x86:pp\xc3\x06\xbfZ\x95\xa1\xf0\x0c\t\x80\x03}q\x00(X\x04\x00\x00\x00slotq\x01K\x01X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x00X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x13\x00\x00\x00put('movie','star')q\x08uu."}}, 'client_request_message': {'content': {'is_retry': False, 'operation': {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}}, 'signed_content': b"l (Y\xf8\xf5\x0c\xaak~S\xfa\xfa\xb1\x97gJ%\xf2\xd1\xa3\x97t\x8c\xf4<\xd9E\x10e%\xec\xbc6\x02q\xdceN_m\x1d!\x8f\xaf/[\n! \xc8\xff\r_0\xd8\xba\x8fq\r`\xca;\r\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x00X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x13\x00\x00\x00put('movie','star')q\x08uu."}}, {'slot': 2, 'operation': {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}, 'order_statements': {'r0': {'order_statement': {'slot': 2, 'operation': {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_order_statement': b"l9\x97\x91M\x02\xf2\x0c\x855\x0c\x03\xe3\xbd\\\x1em\x16\xbb\xd9\xc13\x82p8\x16\xcb\x1eZ\xe4\xa8\x83\xf5\x12\xf1\xd1\x17g\xf2\x12\xd5\xb1\x90{\xe5\xb4\xd6\x854\xe9\x046\xd8\xf3\xb3\x15\xf6>\xc2\x92k.\xfc\x0b\x80\x03}q\x00(X\x04\x00\x00\x00slotq\x01K\x02X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x01X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}}, 'client_request_message': {'content': {'is_retry': False, 'operation': {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"|>\xd0\x05 d!\n$3\xcb8\xea\xc5\\\x99\xa0\xc9e\xdb=\x01\x10\xb5|\x1f[\x81\x03\x9a\x8e \xb6\xb4\xb2\x82A\x85{Fj\x19]C@\xfaA{`\xbd\xcf\x0e\xd6\xd6F\xe4\xb7'rn\r\xb5\x1a\n\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x01X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}}, {'slot': 3, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}, 'order_statements': {'r0': {'order_statement': {'slot': 3, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_order_statement': b"\xc9\xe2\xf9\n \x0b{\xab7\x9b\x18\x8fD\xda\xc1h\n\t\xde\xa9]\xce(\xed\x87^\xe8\xc3\x08V06\xfb\x939'\\!V\x13G=\xaec6\xe3\x9f+u\xd3\r\xbb\x8d\xb8t\xcd\x12e\xfe\xcf\x17\xb7\xe6\x03\x80\x03}q\x00(X\x04\x00\x00\x00slotq\x01K\x03X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append('movie', '123')q\x08uu."}}, 'client_request_message': {'content': {'is_retry': False, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_content': b'\x17\xa7\xeb\xb2/\r\xefi,\xb1\xba\xb0\x8a\xa48k\xf1\xdb\xb9\x1f\xcf\x17\xb8{\xf4\xb0\xfdR!\x15<\xb5n%\x92p\xf0T%:F\xcb\xd7\x8e\xdeg\xcf\xd3!\x15{\n\xf4\x00\x12j"\xa6f\xacs\x9bg\x0e\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append(\'movie\', \'123\')q\x08uu.'}}, {'slot': 3, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}, 'order_statements': {'r0': {'order_statement': {'slot': 3, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_order_statement': b"\xc9\xe2\xf9\n \x0b{\xab7\x9b\x18\x8fD\xda\xc1h\n\t\xde\xa9]\xce(\xed\x87^\xe8\xc3\x08V06\xfb\x939'\\!V\x13G=\xaec6\xe3\x9f+u\xd3\r\xbb\x8d\xb8t\xcd\x12e\xfe\xcf\x17\xb7\xe6\x03\x80\x03}q\x00(X\x04\x00\x00\x00slotq\x01K\x03X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append('movie', '123')q\x08uu."}}, 'client_request_message': {'content': {'is_retry': True, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_content': b"\x11m\xd4\xe6`i\xf88\tj\x11Q\xa07]\x8a\xb9\xcc\x02\xec\x1b\xd0\xa6;\xf7\xa0\xcd\x0e\xe1\x7f.\xde\x10\xed#\xae\xec\xb2\xe5\x02\x8b\x92\xf1?\xf2\x8e$\xf5\xd22I\x90\x10\xd3\x92b\x08\x148]\xbb\xf2N\n\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x88X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append('movie', '123')q\x08uu."}}]  from= r0
[2017-11-21 01:41:32,599] olympus.Olympus<Olympus:8f002>:OUTPUT: Received caught up message from replica: r2
[2017-11-21 01:41:32,602] olympus.Olympus<Olympus:8f002>:OUTPUT: Received caught up message from replica: r3
[2017-11-21 01:41:32,606] olympus.Olympus<Olympus:8f002>:OUTPUT: Received caught up message from replica: r1
[2017-11-21 01:41:32,609] olympus.Olympus<Olympus:8f002>:OUTPUT: Caught up quorum is satisfied.
[2017-11-21 01:41:32,612] olympus.Olympus<Olympus:8f002>:OUTPUT: Sending get_running_state message to:  r2
[2017-11-21 01:41:32,613] replica.Replica<Replica:a0004>:OUTPUT: Received get running state message from Olympus, in = r2
[2017-11-21 01:41:32,615] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('get_running_state', 'pre')  in replica:  r2
[2017-11-21 01:41:32,615] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  get_running_state  found in replica:  r2
[2017-11-21 01:41:32,621] replica.Replica<Replica:a0004>:OUTPUT: Checking for failure associated with trigger, application:  ('get_running_state', 'post')  in replica:  r2
[2017-11-21 01:41:32,622] replica.Replica<Replica:a0004>:OUTPUT: No failure associated with trigger:  get_running_state  found in replica:  r2
[2017-11-21 01:41:32,622] replica.Replica<Replica:a0004>:OUTPUT: Sending running state message to Olympus from:  r2
[2017-11-21 01:41:32,624] olympus.Olympus<Olympus:8f002>:OUTPUT: Running state received from:  r2
[2017-11-21 01:41:32,627] olympus.Olympus<Olympus:8f002>:OUTPUT: Valid running state received in:  r2
[2017-11-21 01:41:32,628] olympus.Olympus<Olympus:8f002>:OUTPUT: Reconfiguration started
[2017-11-21 01:41:32,628] olympus.Olympus<Olympus:8f002>:OUTPUT: Terminating replicas..
[2017-11-21 01:41:32,631] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r0
[2017-11-21 01:41:32,633] replica.Replica<Replica:a0002>:OUTPUT: Received abort message from Olympus in replica:  r0
[2017-11-21 01:41:32,635] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r1
[2017-11-21 01:41:32,636] replica.Replica<Replica:a0003>:OUTPUT: Received abort message from Olympus in replica:  r1
[2017-11-21 01:41:32,637] replica.Replica<Replica:a0002>:OUTPUT: Sending aborted message to Olympus from:  r0
[2017-11-21 01:41:32,638] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r2
[2017-11-21 01:41:32,639] replica.Replica<Replica:a0004>:OUTPUT: Received abort message from Olympus in replica:  r2
[2017-11-21 01:41:32,640] replica.Replica<Replica:a0003>:OUTPUT: Sending aborted message to Olympus from:  r1
[2017-11-21 01:41:32,644] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r3
[2017-11-21 01:41:32,645] replica.Replica<Replica:a0004>:OUTPUT: Sending aborted message to Olympus from:  r2
[2017-11-21 01:41:32,645] replica.Replica<Replica:a0005>:OUTPUT: Received abort message from Olympus in replica:  r3
[2017-11-21 01:41:32,647] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r4
[2017-11-21 01:41:32,648] replica.Replica<Replica:a0006>:OUTPUT: Received abort message from Olympus in replica:  r4
[2017-11-21 01:41:32,649] olympus.Olympus<Olympus:8f002>:OUTPUT: Aborted message received from replica:  r0
[2017-11-21 01:41:32,649] replica.Replica<Replica:a0005>:OUTPUT: Sending aborted message to Olympus from:  r3
[2017-11-21 01:41:32,653] replica.Replica<Replica:a0006>:OUTPUT: Sending aborted message to Olympus from:  r4
[2017-11-21 01:41:32,653] olympus.Olympus<Olympus:8f002>:OUTPUT: Aborted message received from replica:  r1
[2017-11-21 01:41:32,658] olympus.Olympus<Olympus:8f002>:OUTPUT: Aborted message received from replica:  r2
[2017-11-21 01:41:32,662] olympus.Olympus<Olympus:8f002>:OUTPUT: Aborted message received from replica:  r3
[2017-11-21 01:41:32,665] olympus.Olympus<Olympus:8f002>:OUTPUT: Aborted message received from replica:  r4
[2017-11-21 01:41:32,669] olympus.Olympus<Olympus:8f002>:OUTPUT: Received all aborted messages
[2017-11-21 01:41:32,674] olympus.Olympus<Olympus:8f002>:OUTPUT: Initializing cluster with configuration version:  1
[2017-11-21 01:41:33,858] replica.Replica<Replica:a0007>:OUTPUT: Spawned and initialized replica:  r0
[2017-11-21 01:41:33,863] replica.Replica<Replica:a0008>:OUTPUT: Spawned and initialized replica:  r1
[2017-11-21 01:41:33,869] replica.Replica<Replica:a0009>:OUTPUT: Spawned and initialized replica:  r2
[2017-11-21 01:41:33,876] replica.Replica<Replica:a000a>:OUTPUT: Spawned and initialized replica:  r3
[2017-11-21 01:41:33,883] replica.Replica<Replica:a000b>:OUTPUT: Spawned and initialized replica:  r4
[2017-11-21 01:41:33,886] olympus.Olympus<Olympus:8f002>:OUTPUT: Olympus sent verified, potentially undelivered result to client: c0
[2017-11-21 01:41:33,889] client.Client<Client:a0001>:OUTPUT: Verified result message received from Olympus
[2017-11-21 01:41:33,892] client.Client<Client:a0001>:OUTPUT: Not a pending request (1, 'c0')
[2017-11-21 01:41:35,449] client.Client<Client:a0001>:OUTPUT: Wait for result timed out for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:35,451] client.Client<Client:a0001>:OUTPUT: Config requested from Olympus by client= c0
[2017-11-21 01:41:35,456] client.Client<Client:a0001>:OUTPUT: Sending config request from client:  c0
[2017-11-21 01:41:35,461] olympus.Olympus<Olympus:8f002>:OUTPUT: Config request received from client= c0
[2017-11-21 01:41:35,462] replica.Replica<Replica:a0006>:OUTPUT: Result Shuttle Timeout in Replica:  r4
[2017-11-21 01:41:35,470] replica.Replica<Replica:a0006>:OUTPUT: Sent reconfiguration request to Olympus from : r4
[2017-11-21 01:41:35,471] replica.Replica<Replica:a0006>:OUTPUT: Shutting down replica:  r4
[2017-11-21 01:41:35,481] olympus.Olympus<Olympus:8f002>:OUTPUT: Config has been sent to client= c0
[2017-11-21 01:41:35,478] replica.Replica<Replica:a0005>:OUTPUT: Result Shuttle Timeout in Replica:  r3
[2017-11-21 01:41:35,497] client.Client<Client:a0001>:OUTPUT: The config was fetched from Olympus: {'id': 'c0', 'workload': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"], 'private_key': b'efb33292a38912bb3133d8854b24c9981758aa1b01c51d73ff8d324efc5442fa', 'public_keys': {'o1': b'3a86cbbd8d7ff05c991522c1911c5443845c69e78738faec469b778298eefaae', 'c0': b'1384e1ecff6efd11bd538890f04775a720865b9542b5815bfb0fe4d503d15e32', 'c1': b'198fecb429e9ad41d927090abeb2abb6c9522d4880505b52a1cfaea09df0d065', 'r0': b'9c1e7a6b77eadd5dad07cab65ffc28a94d633758afb4d732bd619cdc5366babd', 'r1': b'a11a38450c74a29ecf905bc849b1eb06e9848e5428dd926355f3867dcb369c8a', 'r2': b'a9e8a83c9bd812810130fa726dffde1fd70f40f26be298270a53bdfe62a61812', 'r3': b'88fdb187a2dec3a846d355235b4a785db120e345eb80c20ff026a1d05167f2e4', 'r4': b'a261bf6028619ecff790fb3a49bb3b5caf0ca1af8a1a2d846ec8375e8cb6baf0'}, 'client_timeout': 3.0, 'olympus': {'id': 'o1', 'process_id': <Olympus:8f002>}, 'replicas': {'r0': <Replica:a0007>, 'r1': <Replica:a0008>, 'r2': <Replica:a0009>, 'r3': <Replica:a000a>, 'r4': <Replica:a000b>}, 'head_id': 'r0', 't': 2}
[2017-11-21 01:41:35,503] replica.Replica<Replica:a0005>:OUTPUT: Sent reconfiguration request to Olympus from : r3
[2017-11-21 01:41:35,501] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r4
[2017-11-21 01:41:35,500] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,499] replica.Replica<Replica:a0004>:OUTPUT: Result Shuttle Timeout in Replica:  r2
[2017-11-21 01:41:35,508] client.Client<Client:a0001>:OUTPUT: Retransmitting request to all replicas for operation= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:35,513] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,509] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r3
[2017-11-21 01:41:35,518] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r0
[2017-11-21 01:41:35,520] replica.Replica<Replica:a0008>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r1
[2017-11-21 01:41:35,523] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,524] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r1
[2017-11-21 01:41:35,525] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,525] replica.Replica<Replica:a000a>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r3
[2017-11-21 01:41:35,525] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r1
[2017-11-21 01:41:35,526] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r3
[2017-11-21 01:41:35,527] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r3
[2017-11-21 01:41:35,527] replica.Replica<Replica:a000b>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r4
[2017-11-21 01:41:35,528] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r4
[2017-11-21 01:41:35,529] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r4
[2017-11-21 01:41:35,529] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (2, 'c0')
[2017-11-21 01:41:35,537] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:35,538] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,539] replica.Replica<Replica:a000a>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r4 c0 2 r3
[2017-11-21 01:41:35,539] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r3
[2017-11-21 01:41:35,540] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r3
[2017-11-21 01:41:35,542] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:35,543] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 2 r1
[2017-11-21 01:41:35,543] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,543] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:35,543] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:35,544] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,547] replica.Replica<Replica:a0007>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:35,515] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,516] replica.Replica<Replica:a0004>:OUTPUT: Sent reconfiguration request to Olympus from : r2
[2017-11-21 01:41:35,522] replica.Replica<Replica:a0009>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r2
[2017-11-21 01:41:35,522] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,552] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,559] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r2
[2017-11-21 01:41:35,554] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r2
[2017-11-21 01:41:35,561] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:35,562] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,553] replica.Replica<Replica:a0002>:OUTPUT: Result Shuttle Timeout in head:  r0
[2017-11-21 01:41:35,562] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r2
[2017-11-21 01:41:35,560] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,567] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,567] replica.Replica<Replica:a0002>:OUTPUT: Sent reconfiguration request to Olympus from : r0
[2017-11-21 01:41:35,569] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,573] replica.Replica<Replica:a0009>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:35,571] replica.Replica<Replica:a0002>:OUTPUT: Shutting down replica:  r0
[2017-11-21 01:41:35,573] replica.Replica<Replica:a0008>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
 'pre')  in replica:  r2
[2017-11-21 01:41:35,574] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:35,572] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,582] replica.Replica<Replica:a0009>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:35,573] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r0
application:  ('forwarded_request', 'pre')  in replica:  r2
[2017-11-21 01:41:35,583] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r2
[2017-11-21 01:41:35,579] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,574] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:35,585] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,585] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:35,588] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,592] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 2 r2
[2017-11-21 01:41:35,589] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,594] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:35,594] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,594] replica.Replica<Replica:a0003>:OUTPUT: Result Shuttle Timeout in Replica:  r1
[2017-11-21 01:41:35,597] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:35,598] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,598] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:35,595] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,602] replica.Replica<Replica:a0007>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:35,599] replica.Replica<Replica:a0008>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:35,600] replica.Replica<Replica:a0003>:OUTPUT: Sent reconfiguration request to Olympus from : r1
[2017-11-21 01:41:35,603] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:35,604] replica.Replica<Replica:a0003>:OUTPUT: Shutting down replica:  r1
[2017-11-21 01:41:35,604] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:35,605] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,610] olympus.Olympus<Olympus:8f002>:ERROR: A signature mismatch occurred from source: r1
[2017-11-21 01:41:35,614] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:35,615] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,615] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:35,616] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:35,616] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,619] replica.Replica<Replica:a0007>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:35,622] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 2 r3
[2017-11-21 01:41:35,615] replica.Replica<Replica:a0008>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:35,623] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:35,624] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,623] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r1
[2017-11-21 01:41:35,624] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r1
[2017-11-21 01:41:35,649] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:35,650] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:35,650] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('forwarded_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,651] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,651] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  forwarded_request  found in replica:  r0
[2017-11-21 01:41:35,656] replica.Replica<Replica:a0007>:OUTPUT: Awaiting result shuttle from downstream _replicas
[2017-11-21 01:41:35,657] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 2 r4
[2017-11-21 01:41:35,658] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:35,660] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,697] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:35,698] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,702] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 2 r3
[2017-11-21 01:41:35,702] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:35,702] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:35,702] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,702] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:35,707] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:35,724] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:35,725] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:35,727] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"} OK
[2017-11-21 01:41:35,727] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= OK {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:41:35,728] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:35,728] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:35,731] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'\xa1u\x15\xcc\xefL\x16\xda\x10\xf8\x00FZ\xba\x90\x8e\x8cs\xb7\x15^"v!\xfdc\xd3\x00#\x0f\x1bc\x1b0\xae\xfe\x9d\x13\x03\xc1@\xd4\xee`\xa1\x92\x1c\xd2\x064\x81i\xbf+\x91\xcb\r\t|\xdcmr\xb4\x08\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x03X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:41:35,731] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 2 r2
[2017-11-21 01:41:35,732] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 3 r0
[2017-11-21 01:41:35,732] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:35,733] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,733] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:35,733] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,737] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (3, 'c0')
[2017-11-21 01:41:35,743] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:35,744] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,747] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 3 r1
[2017-11-21 01:41:35,748] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:35,748] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,756] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:35,757] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:35,765] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:35,766] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,768] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 3 r2
[2017-11-21 01:41:35,769] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:35,769] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,770] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:41:35,770] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:35,770] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:35,788] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:35,788] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,791] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:35,791] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:35,791] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 3 r3
[2017-11-21 01:41:35,792] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:35,792] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,795] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 2 r0
[2017-11-21 01:41:35,796] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:35,796] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:35,816] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:35,817] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,820] replica.Replica<Replica:a0007>:OUTPUT: Result received at head, now forwarding to client= c0
[2017-11-21 01:41:35,822] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 3 r4
[2017-11-21 01:41:35,823] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:35,823] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,823] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:35,824] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,830] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r0
[2017-11-21 01:41:35,833] client.Client<Client:a0001>:OUTPUT: Not a pending request (2, 'c0')
[2017-11-21 01:41:35,857] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:35,857] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,861] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 3 r3
[2017-11-21 01:41:35,861] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:35,862] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:35,862] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:35,862] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:35,865] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:35,885] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:35,885] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:35,888] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:35,889] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:35,889] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:35,890] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:35,891] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 3 r2
[2017-11-21 01:41:35,891] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:35,892] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:35,892] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"@\t\r>\xf1^\x9b;\x88E\xffb&d\xaa\x8e\x15~Edq\x0c\x95D0\xe1\x1b\t@x\x83\xber~7C~\x1d\xc1\xf1\xe8\x16Cj=\xc5\x0b\x9e\xdbY\x83\x0f\xe7\x1ex\x93\xb7C\xb9\xddW'\x96\x05\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x04X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:35,894] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 4 r0
[2017-11-21 01:41:35,895] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:35,895] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,899] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (4, 'c0')
[2017-11-21 01:41:35,906] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:35,907] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:35,910] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 4 r1
[2017-11-21 01:41:35,911] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:35,912] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,917] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:35,917] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:35,927] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:35,928] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:35,931] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 3 r1
[2017-11-21 01:41:35,931] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 4 r2
[2017-11-21 01:41:35,931] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:35,932] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:35,932] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:35,932] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,951] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:35,951] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:35,955] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 4 r3
[2017-11-21 01:41:35,955] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:35,956] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,956] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:35,956] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:35,961] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 3 r0
[2017-11-21 01:41:35,962] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:35,962] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:35,979] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:35,980] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:35,983] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 4 r4
[2017-11-21 01:41:35,984] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:35,984] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,015] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,016] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,019] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 4 r3
[2017-11-21 01:41:36,019] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,019] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,020] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,020] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,022] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,043] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,044] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,045] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,045] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,045] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,046] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,047] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'O\xb8\xa5\x075q\x8a\xc1\x94\x1a\x94,>\x9f\xd9_s\xe5\xf2-{\xf5\x12!\x06\x1b+o\xe4\x8f\x19\xb3\xe4G_U\xc2!/\xbfl\xba\xfe#:!Lc9LT\xce\xa2Q\xd4"\x19\xf9g\xd2\x9e>\x88\x00\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x05X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:41:36,049] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 5 r0
[2017-11-21 01:41:36,049] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,049] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,050] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 4 r2
[2017-11-21 01:41:36,050] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,050] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,053] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (5, 'c0')
[2017-11-21 01:41:36,058] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,059] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,062] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 5 r1
[2017-11-21 01:41:36,063] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,063] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,072] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,072] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,077] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,078] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,081] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 5 r2
[2017-11-21 01:41:36,082] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,082] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,082] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 4 r1
[2017-11-21 01:41:36,083] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,083] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,103] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,104] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,106] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,107] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,107] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 5 r3
[2017-11-21 01:41:36,108] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,108] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,111] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 4 r0
[2017-11-21 01:41:36,112] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,112] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,128] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,129] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,132] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 5 r4
[2017-11-21 01:41:36,133] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,133] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,165] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,165] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,169] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 5 r3
[2017-11-21 01:41:36,169] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,170] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,170] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,170] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,173] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,195] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,195] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,196] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,196] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,196] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,196] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,199] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'\xd2\xd3h\xae\xcd\xe9\xa4L\xa2\x1aQI\x1dN\x9cUV\x1d\xd6\xa7\x8d\x9b@:\xc9\x1f\x17\x88\x1ewz!\\\x9ajV\x14\x1cjz\xb1?\xd6\xf8\x86\x14R\x01\xebs2:\x1e;B\xe0\\"\xc8\xa3Z\x97\x0e\x01\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x06X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:41:36,200] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 6 r0
[2017-11-21 01:41:36,200] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 5 r2
[2017-11-21 01:41:36,200] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,201] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,201] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,201] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,205] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (6, 'c0')
[2017-11-21 01:41:36,212] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,213] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,215] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 6 r1
[2017-11-21 01:41:36,216] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,216] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,226] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,226] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,233] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,234] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,237] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 6 r2
[2017-11-21 01:41:36,237] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 5 r1
[2017-11-21 01:41:36,237] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,237] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,238] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,238] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,260] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,260] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,260] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,261] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,264] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 6 r3
[2017-11-21 01:41:36,264] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,265] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,265] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 5 r0
[2017-11-21 01:41:36,266] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,266] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,289] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,289] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,292] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 6 r4
[2017-11-21 01:41:36,293] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,293] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,324] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,325] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,328] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,328] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 6 r3
[2017-11-21 01:41:36,329] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,329] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,329] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,332] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,353] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,354] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,355] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,355] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,356] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,357] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,359] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 6 r2
[2017-11-21 01:41:36,360] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,360] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,362] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 7 r0
[2017-11-21 01:41:36,361] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b']\x84\x04\x84t\xc7+\x0b\x0cT\xd5b\xe0\xd9\xc2\xe0AU\x17\x8e\x85\xa9I\x03"\x08\x8f\xc4\xb2\x07\xcf\xdd\xe2\x9f\x03"<\xa02\xdb^\x04\x8a\x12\x90\xbc\xf3c\x89\xb0\xaa\xc4FaZVj7]\xdb\x0eZ\x8b\t\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x07X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:41:36,364] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,364] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,368] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (7, 'c0')
[2017-11-21 01:41:36,375] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,377] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,381] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 7 r1
[2017-11-21 01:41:36,382] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,382] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,386] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,386] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,400] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,400] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,403] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 6 r1
[2017-11-21 01:41:36,403] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 7 r2
[2017-11-21 01:41:36,404] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,404] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,404] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,405] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,423] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,424] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,424] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,425] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,428] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 7 r3
[2017-11-21 01:41:36,429] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 6 r0
[2017-11-21 01:41:36,430] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,430] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,430] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,430] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,451] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,452] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,456] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 7 r4
[2017-11-21 01:41:36,456] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,457] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,463] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r1
[2017-11-21 01:41:36,462] replica.Replica<Replica:a0007>:OUTPUT: Head initiated checkpointing and sent checkpoint shuttle for slot : 5  from:  r0  to: <Replica:a0008>
[2017-11-21 01:41:36,463] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r1
[2017-11-21 01:41:36,464] replica.Replica<Replica:a0008>:OUTPUT: Received checkpoint shuttle from, in = r0 r1
[2017-11-21 01:41:36,477] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r1
[2017-11-21 01:41:36,477] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r1
[2017-11-21 01:41:36,480] replica.Replica<Replica:a0008>:OUTPUT: Replica sent checkpoint shuttle for slot : 5  from:  r1
[2017-11-21 01:41:36,481] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r2
[2017-11-21 01:41:36,481] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r2
[2017-11-21 01:41:36,481] replica.Replica<Replica:a0009>:OUTPUT: Received checkpoint shuttle from, in = r1 r2
[2017-11-21 01:41:36,494] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r2
[2017-11-21 01:41:36,495] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r2
[2017-11-21 01:41:36,496] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,496] replica.Replica<Replica:a0009>:OUTPUT: Replica sent checkpoint shuttle for slot : 5  from:  r2
[2017-11-21 01:41:36,497] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,498] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r3
[2017-11-21 01:41:36,498] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r3
[2017-11-21 01:41:36,499] replica.Replica<Replica:a000a>:OUTPUT: Received checkpoint shuttle from, in = r2 r3
[2017-11-21 01:41:36,503] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,504] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,507] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,516] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r3
[2017-11-21 01:41:36,518] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r3
[2017-11-21 01:41:36,519] replica.Replica<Replica:a000a>:OUTPUT: Replica sent checkpoint shuttle for slot : 5  from:  r3
[2017-11-21 01:41:36,521] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r4
[2017-11-21 01:41:36,521] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 7 r3
[2017-11-21 01:41:36,521] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r4
[2017-11-21 01:41:36,521] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,522] replica.Replica<Replica:a000b>:OUTPUT: Received checkpoint shuttle from, in = r3 r4
[2017-11-21 01:41:36,522] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,529] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,530] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,531] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,531] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,536] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 8 r0
[2017-11-21 01:41:36,537] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,538] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,535] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"4\xcf\x87 \xe7w\xbc\xbc\xcb\xff\xe7\x89\xea\xa4\xb6\xa6\x7f\xd3\xa1&q]\xc5\x8c\x80\xea\x85\xa3\x89\xc2\xe1\xd4T\xa7\xdc\x0fY\xc15\xc3OV\xdf\xce\xfbw\x17>\xa5\n\x84\xc1\xe8\xf4\x13+@s\xbfAY~\xfc\x07\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x08X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:36,542] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (8, 'c0')
[2017-11-21 01:41:36,544] replica.Replica<Replica:a000b>:OUTPUT: Applied checkpoint until slot:  5  in:  r4
[2017-11-21 01:41:36,544] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r4
[2017-11-21 01:41:36,544] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r4
[2017-11-21 01:41:36,546] replica.Replica<Replica:a000b>:OUTPUT: Tail sent complete checkpoint shuttle for slot : 5  from:  r4
[2017-11-21 01:41:36,548] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,549] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,551] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 8 r1
[2017-11-21 01:41:36,551] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,552] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,552] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,552] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,557] replica.Replica<Replica:a000a>:OUTPUT: Received complete checkpoint shuttle from, in = r4 r3
[2017-11-21 01:41:36,558] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 7 r2
[2017-11-21 01:41:36,558] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,559] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,558] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r3
[2017-11-21 01:41:36,566] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r3
[2017-11-21 01:41:36,571] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,571] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,585] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,586] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,587] replica.Replica<Replica:a000a>:OUTPUT: Applied checkpoint until slot:  5  in:  r3
[2017-11-21 01:41:36,591] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r3
[2017-11-21 01:41:36,591] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r3
[2017-11-21 01:41:36,591] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,592] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 7 r1
[2017-11-21 01:41:36,592] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,592] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,593] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,595] replica.Replica<Replica:a000a>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 5  from:  r3
[2017-11-21 01:41:36,616] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,616] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,617] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,618] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,620] replica.Replica<Replica:a0009>:OUTPUT: Received complete checkpoint shuttle from, in = r3 r2
[2017-11-21 01:41:36,621] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 8 r3
[2017-11-21 01:41:36,621] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,622] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,622] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r2
[2017-11-21 01:41:36,623] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r2
[2017-11-21 01:41:36,624] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 7 r0
[2017-11-21 01:41:36,624] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,626] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,642] replica.Replica<Replica:a0009>:OUTPUT: Applied checkpoint until slot:  5  in:  r2
[2017-11-21 01:41:36,647] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r2
[2017-11-21 01:41:36,648] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r2
[2017-11-21 01:41:36,650] replica.Replica<Replica:a0009>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 5  from:  r2
[2017-11-21 01:41:36,650] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,651] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,651] replica.Replica<Replica:a0008>:OUTPUT: r1 event counts:  {'wedge_request': -1, 'new_configuration': 0, 'checkpoint': 0, 'completed_checkpoint': -1, 'get_running_state': -1, 'catch_up': -1, 'client_events': {'forwarded_request': {'c0': 2}, 'client_request': {'c0': 0}, 'shuttle': {'c0': 6}, 'result_shuttle': {'c0': 5}}}
[2017-11-21 01:41:36,652] replica.Replica<Replica:a0008>:OUTPUT: Received complete checkpoint shuttle from, in = r2 r1
[2017-11-21 01:41:36,653] replica.Replica<Replica:a0008>:OUTPUT: r1 event counts:  {'wedge_request': -1, 'new_configuration': 0, 'checkpoint': 0, 'completed_checkpoint': -1, 'get_running_state': -1, 'catch_up': -1, 'client_events': {'forwarded_request': {'c0': 2}, 'client_request': {'c0': 0}, 'shuttle': {'c0': 6}, 'result_shuttle': {'c0': 5}}}
[2017-11-21 01:41:36,653] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r1
[2017-11-21 01:41:36,654] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r1
[2017-11-21 01:41:36,655] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 8 r4
[2017-11-21 01:41:36,655] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,655] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,675] replica.Replica<Replica:a0008>:OUTPUT: Applied checkpoint until slot:  5  in:  r1
[2017-11-21 01:41:36,678] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r1
[2017-11-21 01:41:36,678] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r1
[2017-11-21 01:41:36,680] replica.Replica<Replica:a0008>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 5  from:  r1
[2017-11-21 01:41:36,682] replica.Replica<Replica:a0007>:OUTPUT: Received complete checkpoint shuttle from, in = r1 r0
[2017-11-21 01:41:36,682] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r0
[2017-11-21 01:41:36,682] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r0
[2017-11-21 01:41:36,699] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,699] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,700] replica.Replica<Replica:a0007>:OUTPUT: Applied checkpoint until slot:  5  in:  r0
[2017-11-21 01:41:36,700] replica.Replica<Replica:a0007>:OUTPUT: Checkpointing process completed successfully until slot:  5
[2017-11-21 01:41:36,703] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 8 r3
[2017-11-21 01:41:36,703] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,703] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,704] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,704] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,706] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,726] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,727] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,727] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,727] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,728] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,728] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,731] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xb5\xc2kU\xa9\x9e\x07c\x81\x9d\xd8{6\xdb\xe5G\r_\xfc\xdc\x8b\xcbL\xad\xc6\xfd\xac\x1d\xf2\x0fu\xe7\xc7\x1f\xde4\x08\xd4_\xdaK@\x8fZ\xc1g\x99\x95\x85\xcd\x10J`\x94:\x87\x9e\x1cX\x057\x16\xa5\t\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\tX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:36,732] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 8 r2
[2017-11-21 01:41:36,732] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,732] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 9 r0
[2017-11-21 01:41:36,732] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,733] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,733] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,736] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (9, 'c0')
[2017-11-21 01:41:36,743] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,743] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,747] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 9 r1
[2017-11-21 01:41:36,747] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,748] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,753] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,754] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,765] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,766] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,769] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 8 r1
[2017-11-21 01:41:36,769] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 9 r2
[2017-11-21 01:41:36,770] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,770] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,770] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,770] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,789] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,790] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,792] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,792] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,793] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 9 r3
[2017-11-21 01:41:36,794] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,794] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,797] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 8 r0
[2017-11-21 01:41:36,798] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,798] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,816] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,817] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,821] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 9 r4
[2017-11-21 01:41:36,822] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,822] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,853] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,854] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,857] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 9 r3
[2017-11-21 01:41:36,857] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:36,858] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,858] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:36,858] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,861] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:36,879] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,880] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:36,883] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:36,884] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,884] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:36,885] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:36,886] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 9 r2
[2017-11-21 01:41:36,887] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,887] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,888] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xc3d\xcaG\x95\xa5l\x12Es\xa3\x01j\x06Xe\x92\x0c}\xfe\xe0@\xba\x8c\xb0\x89\x06\xc3G\xe4t\xdd\xd7%\r\x8b\xd05x\x07\xcd\xbdJ\xd3\xd7 \x12\n\xc3\x05\xba\xedHP]M\x12\x84\x98\x88\x97!C\x08\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\nX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:36,890] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 10 r0
[2017-11-21 01:41:36,891] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:36,891] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,894] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (10, 'c0')
[2017-11-21 01:41:36,901] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:36,901] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:36,903] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 10 r1
[2017-11-21 01:41:36,904] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,904] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,911] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,912] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:36,920] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,920] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 9 r1
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 10 r2
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,923] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,943] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:36,944] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:36,946] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:36,947] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:36,948] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 10 r3
[2017-11-21 01:41:36,949] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:36,949] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,952] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 9 r0
[2017-11-21 01:41:36,952] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:36,952] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:36,974] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:36,974] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:36,978] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 10 r4
[2017-11-21 01:41:36,979] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:36,979] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,010] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,011] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,014] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 10 r3
[2017-11-21 01:41:37,014] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,015] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,015] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,015] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,018] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:37,038] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,038] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,039] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:37,039] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,040] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,040] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:37,042] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 10 r2
[2017-11-21 01:41:37,042] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xea\xfb\n\x832\x1b/m\x0f\xe6\xe4\xcb\xe1\xcc\x19\xc6\xe7`*\x8eTe*\xd8\xbd\xf1[+\x0b`2\xd7|\xeeGX\xe1\xde\xef\xa0\xe7ST\x94\xe3=\x01\t\x99\x03Pc+\xea\xcb\x08eNQ\xb2\xbaT\x93\x0f\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0bX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:37,043] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,043] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,044] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 11 r0
[2017-11-21 01:41:37,044] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:37,044] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,048] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (11, 'c0')
[2017-11-21 01:41:37,054] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:37,055] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,057] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 11 r1
[2017-11-21 01:41:37,057] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,058] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,066] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,067] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,073] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,074] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,078] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 11 r2
[2017-11-21 01:41:37,078] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 10 r1
[2017-11-21 01:41:37,078] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,078] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,079] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,079] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,101] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,102] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,102] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,102] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,105] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 11 r3
[2017-11-21 01:41:37,105] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,106] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,106] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 10 r0
[2017-11-21 01:41:37,107] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:37,107] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:37,128] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,129] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,134] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 11 r4
[2017-11-21 01:41:37,135] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:37,135] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,168] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,168] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,172] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 11 r3
[2017-11-21 01:41:37,172] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,172] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,172] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,172] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,175] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:37,196] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,197] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,198] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:37,198] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,199] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,199] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:37,203] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 11 r2
[2017-11-21 01:41:37,203] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,204] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,204] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xb3n@\x1cN\x1a\xf7gS\xd5\xd116\x0cp\x81\x16!\xd5P\x1f\xb6\x0c\x8fx\x98\x88\x81\xb1]\xa9/\x0c\xfb\x86\xcc\x95<\x92R\xb2\xc1a\xd1\xfb.L\x9b\xd5C\xac\xb3\xadm\xf2\xf6\xc34\xdc\xd3\xa9\xfe\x02\x07\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0cX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:37,205] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 12 r0
[2017-11-21 01:41:37,206] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:37,207] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,210] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (12, 'c0')
[2017-11-21 01:41:37,217] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:37,218] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,220] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 12 r1
[2017-11-21 01:41:37,221] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,221] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,227] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,228] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,239] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,240] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,244] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 12 r2
[2017-11-21 01:41:37,244] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,244] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 11 r1
[2017-11-21 01:41:37,245] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,245] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,245] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,263] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,263] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,268] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 12 r3
[2017-11-21 01:41:37,269] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,269] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,271] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,271] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,276] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 11 r0
[2017-11-21 01:41:37,277] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:37,277] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:37,289] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,289] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,294] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 12 r4
[2017-11-21 01:41:37,295] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:37,295] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,302] replica.Replica<Replica:a0007>:OUTPUT: Head initiated checkpointing and sent checkpoint shuttle for slot : 10  from:  r0  to: <Replica:a0008>
[2017-11-21 01:41:37,303] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r1
[2017-11-21 01:41:37,304] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r1
[2017-11-21 01:41:37,304] replica.Replica<Replica:a0008>:OUTPUT: Received checkpoint shuttle from, in = r0 r1
[2017-11-21 01:41:37,315] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r1
[2017-11-21 01:41:37,315] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r1
[2017-11-21 01:41:37,318] replica.Replica<Replica:a0008>:OUTPUT: Replica sent checkpoint shuttle for slot : 10  from:  r1
[2017-11-21 01:41:37,319] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r2
[2017-11-21 01:41:37,319] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r2
[2017-11-21 01:41:37,320] replica.Replica<Replica:a0009>:OUTPUT: Received checkpoint shuttle from, in = r1 r2
[2017-11-21 01:41:37,332] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r2
[2017-11-21 01:41:37,332] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r2
[2017-11-21 01:41:37,334] replica.Replica<Replica:a0009>:OUTPUT: Replica sent checkpoint shuttle for slot : 10  from:  r2
[2017-11-21 01:41:37,336] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r3
[2017-11-21 01:41:37,336] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r3
[2017-11-21 01:41:37,337] replica.Replica<Replica:a000a>:OUTPUT: Received checkpoint shuttle from, in = r2 r3
[2017-11-21 01:41:37,343] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,344] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,350] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,350] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,353] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'post')  in replica:  r3
[2017-11-21 01:41:37,353] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:37,354] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r3
[2017-11-21 01:41:37,355] replica.Replica<Replica:a000a>:OUTPUT: Replica sent checkpoint shuttle for slot : 10  from:  r3
[2017-11-21 01:41:37,357] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('checkpoint', 'pre')  in replica:  r4
[2017-11-21 01:41:37,357] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  checkpoint  found in replica:  r4
[2017-11-21 01:41:37,358] replica.Replica<Replica:a000b>:OUTPUT: Received checkpoint shuttle from, in = r3 r4
[2017-11-21 01:41:37,358] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 12 r3
[2017-11-21 01:41:37,359] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,364] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,375] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:37,375] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,376] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,376] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:37,381] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 13 r0
[2017-11-21 01:41:37,383] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:37,383] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,380] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\x1e]c\x94\xb1\xa7\x826\x1do\xda_\xa0.>\x1a\xb5P\xad\x15\xa8\xbe\xd0\x99\x16\xd4_\x07\x08\xa2\xe9\x80x*sT;R\x00\xfa{\xf4\x8c\xc2\xc02\xe7\xf0\xa56\xb8\xef\x81\xcf\xfb\xca\x92\xce\xeb\x99-C\x98\x04\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\rX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:41:37,385] replica.Replica<Replica:a000b>:OUTPUT: Applied checkpoint until slot:  10  in:  r4
[2017-11-21 01:41:37,388] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (13, 'c0')
[2017-11-21 01:41:37,388] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r4
[2017-11-21 01:41:37,389] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r4
[2017-11-21 01:41:37,392] replica.Replica<Replica:a000b>:OUTPUT: Tail sent complete checkpoint shuttle for slot : 10  from:  r4
[2017-11-21 01:41:37,395] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:37,395] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,397] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 13 r1
[2017-11-21 01:41:37,397] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,398] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,398] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,398] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,402] replica.Replica<Replica:a000a>:OUTPUT: Received complete checkpoint shuttle from, in = r4 r3
[2017-11-21 01:41:37,403] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r3
[2017-11-21 01:41:37,403] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 12 r2
[2017-11-21 01:41:37,404] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,405] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,404] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r3
[2017-11-21 01:41:37,414] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,415] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,428] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,428] replica.Replica<Replica:a000a>:OUTPUT: Applied checkpoint until slot:  10  in:  r3
[2017-11-21 01:41:37,428] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,431] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r3
[2017-11-21 01:41:37,433] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 13 r2
[2017-11-21 01:41:37,432] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r3
[2017-11-21 01:41:37,433] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,434] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 12 r1
[2017-11-21 01:41:37,434] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,435] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,435] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,437] replica.Replica<Replica:a000a>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 10  from:  r3
[2017-11-21 01:41:37,457] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,457] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,460] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,460] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,461] replica.Replica<Replica:a0009>:OUTPUT: Received complete checkpoint shuttle from, in = r3 r2
_id)  r2 c0 13 r3
[2017-11-21 01:41:37,462] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,462] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r2
[2017-11-21 01:41:37,462] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,463] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r2
[2017-11-21 01:41:37,466] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 12 r0
[2017-11-21 01:41:37,467] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:37,467] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:37,487] replica.Replica<Replica:a0009>:OUTPUT: Applied checkpoint until slot:  10  in:  r2
[2017-11-21 01:41:37,491] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r2
[2017-11-21 01:41:37,491] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,491] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r2
[2017-11-21 01:41:37,492] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,493] replica.Replica<Replica:a0009>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 10  from:  r2
[2017-11-21 01:41:37,494] replica.Replica<Replica:a0008>:OUTPUT: r1 event counts:  {'wedge_request': -1, 'new_configuration': 0, 'checkpoint': 1, 'completed_checkpoint': 0, 'get_running_state': -1, 'catch_up': -1, 'client_events': {'forwarded_request': {'c0': 2}, 'client_request': {'c0': 0}, 'shuttle': {'c0': 11}, 'result_shuttle': {'c0': 10}}}
[2017-11-21 01:41:37,495] replica.Replica<Replica:a0008>:OUTPUT: Received complete checkpoint shuttle from, in = r2 r1
[2017-11-21 01:41:37,495] replica.Replica<Replica:a0008>:OUTPUT: r1 event counts:  {'wedge_request': -1, 'new_configuration': 0, 'checkpoint': 1, 'completed_checkpoint': 0, 'get_running_state': -1, 'catch_up': -1, 'client_events': {'forwarded_request': {'c0': 2}, 'client_request': {'c0': 0}, 'shuttle': {'c0': 11}, 'result_shuttle': {'c0': 10}}}
[2017-11-21 01:41:37,495] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 13 r4
[2017-11-21 01:41:37,496] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r1
[2017-11-21 01:41:37,496] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:37,496] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r1
[2017-11-21 01:41:37,496] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,517] replica.Replica<Replica:a0008>:OUTPUT: Applied checkpoint until slot:  10  in:  r1
[2017-11-21 01:41:37,520] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'post')  in replica:  r1
[2017-11-21 01:41:37,521] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r1
[2017-11-21 01:41:37,522] replica.Replica<Replica:a0008>:OUTPUT: Replica sent complete checkpoint shuttle for slot : 10  from:  r1
[2017-11-21 01:41:37,523] replica.Replica<Replica:a0007>:OUTPUT: Received complete checkpoint shuttle from, in = r1 r0
[2017-11-21 01:41:37,524] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('completed_checkpoint', 'pre')  in replica:  r0
[2017-11-21 01:41:37,524] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  completed_checkpoint  found in replica:  r0
[2017-11-21 01:41:37,540] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,541] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,543] replica.Replica<Replica:a0007>:OUTPUT: Applied checkpoint until slot:  10  in:  r0
[2017-11-21 01:41:37,543] replica.Replica<Replica:a0007>:OUTPUT: Checkpointing process completed successfully until slot:  10
[2017-11-21 01:41:37,545] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 13 r3
[2017-11-21 01:41:37,546] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,546] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,546] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,546] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,549] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:37,570] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,570] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:41:37,570] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,571] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:41:37,571] client.Client<Client:a0001>:OUTPUT: The following operation was created: {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}
[2017-11-21 01:41:37,572] client.Client<Client:a0001>:OUTPUT: return_value= star123
[2017-11-21 01:41:37,575] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 13 r2
[2017-11-21 01:41:37,576] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,576] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,576] client.Client<Client:a0001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}}, 'signed_content': b"7\x12+\x8fm,\xb5\x07}7\xe1\xdd\x9b:\xd9\xe1\n\xddbk\x16\xa8H^\xdc\xc01\xc8\x9b\x07\x88J\xc0\xad,\xc6\x95\xccmW\xa8I\xb74\xab\xbd[y9\x18\x17\xe7\xc1\xfa\x7f\x03\xb2\xca\xcd\x9fEs\xeb\x02\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0eX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\r\x00\x00\x00get('movie');q\x08uu."}
[2017-11-21 01:41:37,577] replica.Replica<Replica:a0007>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 14 r0
[2017-11-21 01:41:37,578] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:41:37,578] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,582] replica.Replica<Replica:a0007>:OUTPUT: Head is processing request (operation_id, client_id) (14, 'c0')
[2017-11-21 01:41:37,589] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:41:37,590] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:41:37,593] replica.Replica<Replica:a0008>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 14 r1
[2017-11-21 01:41:37,594] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,594] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,598] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,599] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,609] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,610] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:41:37,613] replica.Replica<Replica:a0008>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 13 r1
[2017-11-21 01:41:37,613] replica.Replica<Replica:a0009>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 14 r2
[2017-11-21 01:41:37,613] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:41:37,614] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,614] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,614] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,629] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,630] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:41:37,634] replica.Replica<Replica:a0008>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:41:37,635] replica.Replica<Replica:a000a>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r2 c0 14 r3
[2017-11-21 01:41:37,635] replica.Replica<Replica:a0008>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:41:37,636] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,636] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,641] replica.Replica<Replica:a0007>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 13 r0
[2017-11-21 01:41:37,642] replica.Replica<Replica:a0007>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:41:37,642] replica.Replica<Replica:a0007>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:41:37,660] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,660] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r3
[2017-11-21 01:41:37,664] replica.Replica<Replica:a000b>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r3 c0 14 r4
[2017-11-21 01:41:37,664] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r4
[2017-11-21 01:41:37,664] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,695] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,695] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,699] replica.Replica<Replica:a000a>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r4 c0 14 r3
[2017-11-21 01:41:37,699] replica.Replica<Replica:a000b>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r4
[2017-11-21 01:41:37,699] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r3
[2017-11-21 01:41:37,699] replica.Replica<Replica:a000b>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r4
[2017-11-21 01:41:37,699] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,703] client.Client<Client:a0001>:OUTPUT: A result was received from replica= r4
[2017-11-21 01:41:37,721] client.Client<Client:a0001>:OUTPUT: The operation result pair is= {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"} star123
[2017-11-21 01:41:37,721] client.Client<Client:a0001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}
[2017-11-21 01:41:37,722] client.Client<Client:a0001>:OUTPUT: ~~~~~~~~~~Final state: {'movie': 'star123'}
[2017-11-21 01:41:37,722] client.Client<Client:a0001>:OUTPUT: Requesting shared object from head from client: c0
[2017-11-21 01:41:37,724] replica.Replica<Replica:a000a>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r3
[2017-11-21 01:41:37,724] replica.Replica<Replica:a000a>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r3
[2017-11-21 01:41:37,726] replica.Replica<Replica:a0007>:OUTPUT: Received wedge request from Olympus in Replica:  r0
[2017-11-21 01:41:37,728] replica.Replica<Replica:a0009>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r3 c0 14 r2
[2017-11-21 01:41:37,728] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r2
[2017-11-21 01:41:37,729] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,729] replica.Replica<Replica:a0007>:OUTPUT: Received shared_object_request from client= c0
[2017-11-21 01:41:37,732] client.Client<Client:a0001>:OUTPUT: Received shared object from head =  {'movie': 'star123'}
[2017-11-21 01:41:37,736] client.Client<Client:a0001>:OUTPUT: Final results are consistent
[2017-11-21 01:41:37,736] client.Client<Client:a0001>:OUTPUT: Workload finished for client:  c0
[2017-11-21 01:41:37,737] client.Client<Client:a0001>:OUTPUT: Sending done message to Olympus and terminating
[2017-11-21 01:41:37,740] olympus.Olympus<Olympus:8f002>:OUTPUT: Done message received from client:  c0
[2017-11-21 01:41:37,744] olympus.Olympus<Olympus:8f002>:OUTPUT: Terminating replicas..
[2017-11-21 01:41:37,747] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r0
[2017-11-21 01:41:37,749] replica.Replica<Replica:a0007>:OUTPUT: Received abort message from Olympus in replica:  r0
[2017-11-21 01:41:37,751] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r1
[2017-11-21 01:41:37,751] replica.Replica<Replica:a0009>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r2
[2017-11-21 01:41:37,753] replica.Replica<Replica:a0007>:OUTPUT: Sending aborted message to Olympus from:  r0
:  r1
[2017-11-21 01:41:37,754] replica.Replica<Replica:a0009>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r2
[2017-11-21 01:41:37,756] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r2
[2017-11-21 01:41:37,757] replica.Replica<Replica:a0007>:OUTPUT: Shutting down replica:  r0
[2017-11-21 01:41:37,758] replica.Replica<Replica:a0008>:OUTPUT: Sending aborted message to Olympus from:  r1
[2017-11-21 01:41:37,759] replica.Replica<Replica:a0009>:OUTPUT: Received abort message from Olympus in replica:  r2
[2017-11-21 01:41:37,760] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r3
[2017-11-21 01:41:37,761] replica.Replica<Replica:a000a>:OUTPUT: Received abort message from Olympus in replica:  r3
[2017-11-21 01:41:37,762] replica.Replica<Replica:a0008>:OUTPUT: Shutting down replica:  r1
[2017-11-21 01:41:37,763] olympus.Olympus<Olympus:8f002>:OUTPUT: Sent abort command to replica : r4
[2017-11-21 01:41:37,763] olympus.Olympus<Olympus:8f002>:OUTPUT: Olympus terminating
[2017-11-21 01:41:37,764] olympus.Olympus<Olympus:8f002>:OUTPUT: elapsed time (seconds):  13.183407068252563
[2017-11-21 01:41:37,764] replica.Replica<Replica:a000b>:OUTPUT: Received abort message from Olympus in replica:  r4
[2017-11-21 01:41:37,765] replica.Replica<Replica:a000a>:OUTPUT: Sending aborted message to Olympus from:  r3
[2017-11-21 01:41:37,768] replica.Replica<Replica:a000b>:OUTPUT: Sending aborted message to Olympus from:  r4
[2017-11-21 01:41:37,772] replica.Replica<Replica:a000b>:OUTPUT: Shutting down replica:  r4
[2017-11-21 01:41:37,771] replica.Replica<Replica:a000a>:OUTPUT: Shutting down replica:  r3
[2017-11-21 01:41:37,773] replica.Replica<Replica:a0009>:OUTPUT: Sending aborted message to Olympus from:  r2
[2017-11-21 01:41:37,803] replica.Replica<Replica:a0009>:OUTPUT: Shutting down replica:  r2
[2017-11-21 01:41:38,504] replica.Replica<Replica:a0005>:OUTPUT: Result Shuttle Timeout in Replica:  r3
[2017-11-21 01:41:38,511] replica.Replica<Replica:a0005>:OUTPUT: Sent reconfiguration request to Olympus from : r3
[2017-11-21 01:41:38,512] replica.Replica<Replica:a0005>:OUTPUT: Result Shuttle Timeout in Replica:  r3
[2017-11-21 01:41:38,518] replica.Replica<Replica:a0005>:OUTPUT: Sent reconfiguration request to Olympus from : r3
[2017-11-21 01:41:38,519] replica.Replica<Replica:a0005>:OUTPUT: Result Shuttle Timeout in Replica:  r3
[2017-11-21 01:41:38,529] replica.Replica<Replica:a0005>:OUTPUT: Sent reconfiguration request to Olympus from : r3
[2017-11-21 01:41:38,530] replica.Replica<Replica:a0005>:OUTPUT: Shutting down replica:  r3
[2017-11-21 01:41:38,554] replica.Replica<Replica:a0004>:OUTPUT: Result Shuttle Timeout in Replica:  r2
[2017-11-21 01:41:38,561] replica.Replica<Replica:a0004>:OUTPUT: Sent reconfiguration request to Olympus from : r2
[2017-11-21 01:41:38,561] replica.Replica<Replica:a0004>:OUTPUT: Result Shuttle Timeout in Replica:  r2
[2017-11-21 01:41:38,565] replica.Replica<Replica:a0004>:OUTPUT: Sent reconfiguration request to Olympus from : r2
[2017-11-21 01:41:38,566] replica.Replica<Replica:a0004>:OUTPUT: Result Shuttle Timeout in Replica:  r2
[2017-11-21 01:41:38,569] replica.Replica<Replica:a0004>:OUTPUT: Sent reconfiguration request to Olympus from : r2
[2017-11-21 01:41:38,570] replica.Replica<Replica:a0004>:OUTPUT: Result Shuttle Timeout in Replica:  r2
[2017-11-21 01:41:38,574] replica.Replica<Replica:a0004>:OUTPUT: Sent reconfiguration request to Olympus from : r2
[2017-11-21 01:41:38,575] replica.Replica<Replica:a0004>:OUTPUT: Shutting down replica:  r2
